main.go

```diff
@@ -40,59 +40,42 @@ func main() {
 
 	// initialize resolver
 	config := records.SetConfig(*cjson)
-	resolver := resolver.New(version, config)
+	res := resolver.New(version, config)
 	errch := make(chan error)
 
 	// launch DNS server
 	if config.DNSOn {
-		go func() { errch <- <-resolver.LaunchDNS() }()
+		go func() { errch <- <-res.LaunchDNS() }()
 	}
 
 	// launch HTTP server
 	if config.HTTPOn {
-		go func() { errch <- <-resolver.LaunchHTTP() }()
+		go func() { errch <- <-res.LaunchHTTP() }()
 	}
 
-	var newLeader <-chan struct{}
+	var masters <-chan []string
 	var zkErr <-chan error
 	// launch Zookeeper listener
 	if config.Zk != "" {
-		newLeader, zkErr = resolver.LaunchZK(zkInitialDetectionTimeout)
+		masters, zkErr = res.LaunchZK(zkInitialDetectionTimeout)
 		go func() { errch <- <-zkErr }()
 	} else {
 		// uniform behavior when new leader from masters field
-		leader := make(chan struct{}, 1)
-		leader <- struct{}{}
-		newLeader = leader
+		mc := resolver.NewMasterChanger(config.Masters)
```

Comment:
I didn't immediately think about this approach, sorry for only bringing it up now. Feel free to think about it later, but here it goes: What if the caller would provide the channel to be notified to? That way we could make `Emit` private and replace `func (mc MasterChanger) ChangedChan() chan []string` with `func (mc MasterChanger) Subscribe(ch)`
```go
} else {
    masters = make(chan []string, len(config.Masters))
    for _, master := range config.Masters {
        masters <- master
    }
}
```